PWMoutputMATLABtutorial (#222)
arduino/docs-content: content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md

---

> title: 'Controlling PWM output with a potentiometer on the Arduino Uno via MATLAB'

Reviewer: ```suggestion
title: 'Controlling PWM output with a potentiometer on the Arduino® UNO R3 via MATLAB'
```

difficulty: intermediate
compatible-products: [Arduino Uno Rev 3, Potentiometer]

> description: 'Learn how to dim an LED connceted to the Arduino Uno with MATLAB.'

Reviewer: ```suggestion
description: 'Learn how to dim an LED connceted to the UNO R3 with MATLAB.'
```

tags:
  - MATLAB
author: 'Mohammed Hussain Jahangiri'
hardware:
  - hardware/boards/arduino-uno-rev3
software:
  - MATLAB
---

## Introduction

> In this tutorial, we will use MATLAB to turn on an external LED, and then to adjust the output using MATLAB® Support Package for Arduino® Hardware.

Reviewer: ```suggestion
In this tutorial, we will use MATLAB® to turn on an external LED, and then to adjust the output using MATLAB® Support Package for Arduino® Hardware.
```

### Goals

The goals of this project are:

> - Control the Arduino Uno LED with MATLAB.

Reviewer: ```suggestion
- Control the UNO R3 LED with MATLAB.
```

Reviewer: After the first time mention of the full name of the product, it is ok to used the product name without the 'Arduino®'

> - Control the Arduino Uno PWM output with MATLAB.

Reviewer: ```suggestion
- Control the UNO R3 PWM output with MATLAB.
```

- Create a dimmable light using a potentiometer.

### Hardware & Software Needed

> - [Arduino Uno Rev3](https://store.arduino.cc/products/arduino-uno-rev3)

Reviewer: ```suggestion
- [UNO R3](https://store.arduino.cc/products/arduino-uno-rev3)
```

- [USB 2.0 Cable Type A/B](https://store.arduino.cc/products/usb-2-0-cable-type-a-b)
- Any potentiometer connectable to a breadboard
- Valid MATLAB® licence
- [MATLAB® Support Package for Arduino® Hardware](https://www.mathworks.com/matlabcentral/fileexchange/47522-matlab-support-package-for-arduino-hardware)

> ***The Arduino Uno, potentiometer, USB 2.0 Cable Type A/B and jumper wires are included as part of the [Arduino Starter Kit](https://store.arduino.cc/collections/kits/products/arduino-starter-kit-multi-language)***

Reviewer: ```suggestion
***The UNO R3, potentiometer, USB 2.0 Cable Type A/B and jumper wires are included as part of the [Arduino Starter Kit](https://store.arduino.cc/collections/kits/products/arduino-starter-kit-multi-language)***
```

***A valid MATLAB® licence is needed. Your workplace or education institution may have a subscription. Alternatively, a one-year trial subscription to MATLAB® is included as part of the [Arduino Engineering Kit R2](https://store.arduino.cc/products/arduino-engineering-kit-rev2).***

### Why MATLAB?

> MATLAB® is an educational and industrial programming platform used to analyse data, perform simulations and carry out model based designs. Through an interactive communication with an Arduino board, you can expand MATLAB's capabilities, while also gaining access to a wide range of math, engineering and plotting functions. Check out the capabilities of MATLAB® over on the [MathWorks website](https://www.mathworks.com/solutions.html#capabilities). In this tutorial, we will establish the connection with MATLAB via a USB and Wi-Fi connection. The [MATLAB Support Package for Arduino Hardware](https://www.mathworks.com/matlabcentral/fileexchange/47522-matlab-support-package-for-arduino-hardware) must be installed (which may require a MathWorks account), in order to facilitate the communication between the MATLAB software on your computer and your Arduino board. 

Reviewer: Updated the MATLAB® and MathWorks® trademark

 

### LED on the Arduino Uno R3

> The internal LED in the Arduino Uno is connected to Digital Pin 13. By pulling this pin HIGH, we can turn the LED on and by pulling it LOW we can turn the LED off. But to output a variable voltage we have to use pins such as pin ~11 or ~10.

Reviewer: ```suggestion
The internal LED in the UNO R3 is connected to Digital Pin 13. By pulling this pin HIGH, we can turn the LED on and by pulling it LOW we can turn the LED off. But to output a variable voltage we have to use pins such as pin ~11 or ~10.
```

<p align="center">

> <img src="http://electronicswithyou.com/wp-content/uploads/2019/05/Arduino-Uno-LED.jpg"/>

Reviewer: Please use the markdown formatting instead, for example:

![image description](link.com/url/to/image.jpg)

Author: Will do.

</p>

## Control LED over USB with MATLAB
**1.** Connect your board to the computer via the USB 2.0 Cable Type A/B.

![USB connection to computer](assets/Arduino-Connect.png)

**2.** Open MATLAB and run the command `arduinosetup()` in the Command Window.

![arduino Setup](assets/arduinosetup.png)

> **3.** A Graphical User Interface (GUI) appears, that will help you set up the MATLAB-Arduino connection. To establish the connection over the USB cable, make sure that the USB radio box is selected and then click on Next.

Reviewer: ```suggestion
**3.** A Graphical User Interface (GUI) appears, that will help you set up the MATLAB®-Arduino connection. To establish the connection over the USB cable, make sure that the USB radio box is selected and then click on Next.
```

![Selection of Connection Type](assets/connectiontype.png)

> **4.** Choose the "Uno" from the dropdown menu, as well as the relevant COM port, and click on the blue `Program` button to upload the Arduino Server to the board. When you have done so, click on next.

Reviewer: ```suggestion
**4.** Choose the "UNO" from the dropdown menu, as well as the relevant COM port, and click on the blue `Program` button to upload the Arduino Server to the board. When you have done so, click on next.
```

<p align="center">
<img src="https://i.imgur.com/AUhpmWO.png"/>
</p>

> **5.** In this step you can review the connection type, Port, board and loaded libraries. You can also click on Test connection to evaluate the Arduino-MATLAB connection. Next, you should see a Green check-mark signalling the successful connection, as shown in the image below.

Reviewer: ```suggestion
**5.** In this step you can review the connection type, Port, board and loaded libraries. You can also click on Test connection to evaluate the Arduino-MATLAB® connection. Next, you should see a Green check-mark signalling the successful connection, as shown in the image below.
```

<p align="center">
<img src="https://i.imgur.com/QEXZ27s.png"/>
</p>

> **6.** The Arduino Uno is now configured to interact with MATLAB. Now to create the object in MATLAB, we run the command `a=arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.

Reviewer: ```suggestion
**6.** The UNO R3 is now configured to interact with MATLAB®. Now to create the object in MATLAB®, we run the command `a=arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.
```

![](assets/arduinoproperties.png)

> **7.** Enter the command `writeDigitalPin(a,'D13',1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 (D13) is connected to the built-in LED, `a` is the arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.

Reviewer: ```suggestion
**7.** Enter the command `writeDigitalPin(a,'D13',1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 (D13) is connected to the built-in LED, `a` is the Arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.
```

***Unlike in the Arduino IDE, here you cannot address the built-in LED using the `LED_BUILTIN` function.***

**8.** Now, you can use this command to continuously blink an LED as part of a `while` loop. Try entering the script below.

```arduino
while (1)
      writeDigitalPin(a, 'D13', 0);
      pause(0.5);
      writeDigitalPin(a, 'D13', 1);
      pause(0.5);
end
```

**9.** See what happens when you change the pause duration. Try replacing the while with an `if` statement to make it blink a certain number of times.

## Set up the circuit and code

**1.** On one breadboard we have the white jumper wire sending 5 volts to the potentiometer. The orange cable is connected to GND(Ground) and the gray wire is our input to A0. A0 will read the variable voltage coming in from pin13 after being adjusted through the potentiometer.

> On the right breadboard we have a simple LED circuit comprised of a red output wire connected to pin11, a resistor and the LED finally connected to GND via the green wire

Reviewer: ```suggestion
On the right breadboard we have a simple LED circuit comprised of a red output wire connected to pin11, a resistor and the LED finally connected to GND via the green wire.
```

Reviewer: Missing punctuation

![](assets/curcuit.jpg)

***Make sure to connect the longer leg of the LED to the resistor and the shorter to GND.***

> **2.** Set your output voltage to 1 in the previously mentioned way, using `writeDigitalPin(a, 'D13', 1);` Now we'll check to see if A0 is actually reading an adjustable voltage. To see if the input changes as we turn the potentiometer use `readVoltage(a, A0);` in a loop.

Reviewer: ```suggestion
**2.** Set your output voltage to 1 in the previously mentioned way, using `writeDigitalPin(a, 'D13', 1);` Now we'll check to see if A0 is actually reading an adjustable voltage. To see if the input changes as we turn the potentiometer use `readVoltage(a, A0);` in a loop.

```

Reviewer: we should always add a space before each code snippet

<p align="center">
<img src="https://i.imgur.com/jRj6dh1.png"/>
</p>

<p align="center">

> MATLAB should be displaying different amount for "Volts as your turn the potentiometer.

Reviewer: ```suggestion
MATLAB® should be displaying different amount for "Volts as your turn the potentiometer.
```

</p>

> **3.** All that is left is to send the amount recieved from A0 to pin 11. For this we need a different command: `writePWMVolatge()`.

Reviewer: ```suggestion
**3.** All that is left is to send the amount recieved from A0 to pin 11. For this we need a different command: `writePWMVolatge()`.

```

Reviewer: we should always add a space before each code snippet

<p align="center">
<img src="https://gcdnb.pbrd.co/images/PUSyiiyMTi0T.png?o=1"/>
</p>

### Troubleshoot

If the code is not working, there are some common issues we can troubleshoot:

- Make sure to clear previous Arduino objects linked to the device. You can do this by running the command `clear a` where `a` is the Arduino object.
- If you get the message `Unrecognized function or variable 'arduinosetup' ` ensure that you have installed the [MATLAB Support Package for Arduino Hardware](https://www.mathworks.com/matlabcentral/fileexchange/47522-matlab-support-package-for-arduino-hardware).

## Conclusion

You can now control your output using variable inputs to control your devices and make use of the powerful features for scientific computing and developing engineering applications!

## Further Ideas

- You can also use the command line arguments to upload the MATLAB server to the Nano 33 IoT via the `arduino()` command. See more information on using this function in the [MathWorks documentation](https://www.mathworks.com/help/supportpkg/arduinoio/ref/arduino.html).

> 

Reviewer: I suggest to remove the space between the bullet points to make the list coherent

> - Try writing a MATLAB code that slowly increases and then decreases the blink speed instead of adjusting the brightness.

Reviewer: ```suggestion
- Try writing a MATLAB® code that slowly increases and then decreases the blink speed instead of adjusting the brightness.
```

---
title: 'Controlling PWM Output with a Potentiometer via MATLAB®'
difficulty: intermediate
compatible-products: [Arduino Uno Rev 3, Potentiometer]
description: 'Learn how to dim an LED connected to the UNO R3 with MATLAB®.'
tags:
  - MATLAB
author: 'Mohammed Hussain Jahangiri'
hardware:
  - hardware/boards/arduino-uno-rev3
software:
  - MATLAB
---

## Introduction

In this tutorial, we will use MATLAB® to turn on an external LED, and then to adjust the output using MATLAB® Support Package for Arduino® Hardware.

### Goals

The goals of this project are:

- Control the UNO R3 LED with MATLAB.
- Control the UNO R3 PWM output with MATLAB.
- Create a dimmable light using a potentiometer.

### Hardware & Software Needed

- [UNO R3](https://store.arduino.cc/products/arduino-uno-rev3)
- [USB 2.0 Cable Type A/B](https://store.arduino.cc/products/usb-2-0-cable-type-a-b)

> - Any potentiometer connectable to a breadboard

Reviewer 2: ```suggestion
- Potentiometer trough-hole
```

- Valid MATLAB® licence
- [MATLAB® Support Package for Arduino® Hardware](https://www.mathworks.com/matlabcentral/fileexchange/47522-matlab-support-package-for-arduino-hardware)

> ***The UNO R3, potentiometer, USB 2.0 Cable Type A/B and jumper wires are included as part of the [Arduino Starter Kit](https://store.arduino.cc/collections/kits/products/arduino-starter-kit-multi-language)***

Reviewer 2: Not needed
```suggestion
```

 
## Connecting the Board to MATLAB
**1.** Connect your board to the computer via the USB 2.0 Cable Type A/B.

![USB connection to computer](assets/Arduino-Connect1080.png)

**2.** Open MATLAB and run the command `arduinosetup()` in the Command Window.

![Arduino Setup](assets/arduinosetup.png)

**3.** A Graphical User Interface (GUI) appears, that will help you set up the MATLAB®-Arduino connection. To establish the connection over the USB cable, make sure that the USB radio box is selected and then click on Next.

![Selection of Connection Type](assets/connectiontype.png)

**4.** Choose the "UNO" from the dropdown menu, as well as the relevant COM port, and click on the blue `Program` button to upload the Arduino Server to the board. When you have done so, click on next.

![Programming the Arduino](assets/HardwareSetup1080.PNG)

**5.** In this step you can review the connection type, Port, board and loaded libraries. You can also click on Test connection to evaluate the Arduino-MATLAB® connection. Next, you should see a Green check-mark signalling the successful connection, as shown in the image below.

![Programming successful](assets/SetupSuccess1080.PNG)

> **6.** The UNO R3 is now configured to interact with MATLAB®. Now to create the object in MATLAB®, we run the command `a=arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.

Reviewer 2: ```suggestion
**6.** The UNO R3 is now configured to interact with MATLAB®. Now to create the object in MATLAB®, we run the command `a = arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.
```

![Arduino properties](assets/arduinoproperties1080.png)

> **7.** Enter the command `writeDigitalPin(a,'D13',1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 (D13) is connected to the built-in LED, `a` is the Arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.

Reviewer 2: ```suggestion
**7.** Enter the command `writeDigitalPin(a, 'D13', 1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 **D13** is connected to the built-in LED, `a` is the Arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.
```

> ***Unlike in the Arduino IDE, here you cannot address the built-in LED using the `LED_BUILTIN` function.***

Reviewer 2: ```suggestion
***Unlike in the Arduino IDE, here you cannot address the built-in LED using the `LED_BUILTIN` macro.***
```

**8.** Now, you can use this command to continuously blink an LED as part of a `while` loop. Try entering the script below.

> ```arduino

Reviewer 3: ````suggestion
```
````
This code is not written in the Arduino language, so it should not be syntax highlighted according to the rules of that language.

I see that the Linguist library used by GitHub does have a `MATLAB` identifier. I'm not sure which one, if any is used by the Arduino website system though.

https://github.com/github/linguist/blob/v7.20.0/lib/linguist/languages.yml#L3553

while (1)
      writeDigitalPin(a, 'D13', 0);
      pause(0.5);
      writeDigitalPin(a, 'D13', 1);
      pause(0.5);
end
```

> **9.** See what happens when you change the pause duration. Try replacing the while with an `if` statement to make it blink a certain number of times.

Reviewer 2: Confusing for the end user, reword, change or remove
```suggestion
```

## Circuit

On one breadboard there is a white jumper wire sending 5 volts to the potentiometer. The orange cable is connected to GND(Ground) and the gray wire is our input to A0. A0 will read the variable voltage coming in from pin13 after being adjusted through the potentiometer.
On the right breadboard we have a simple LED circuit comprised of a red output wire from pin11, a resistor and the LED finally connected to GND via the green wire.

> ![Circuit](assets/circuit.png)

Reviewer: ```suggestion
![Full circuit of the setup.](assets/circuit.png)
```

Reviewer: @MessiahJa found a last image alternative text here, if you agree with the description - please accept the change 🙏🏼

> ![Full circuit of the setup.](assets/circuit.png)

Reviewer 2: @jhansson-ard We will need to replace this with an schematic, I'll send the draft

> ***Make sure to connect the longer leg of the LED to the resistor and the shorter to GND.***

Reviewer 2: ```suggestion
***Make sure to connect the longer leg of the LED (anode) to the resistor and the shorter (cathode) to GND.***
```

## Programming the Board

> **1.** Set your output to 1 in the previously mentioned way using `writeDigitalPin(a, 'D13', 1);`. As mentioned earlier, 1 means ON or HIGH and sends a 5V current through pin13. Check to see if A0 is actually reading an adjustable voltage using `readVoltage(a, 'A0');` in a loop and then use `display(Volts)` to print the value of `Volts`.

Reviewer 2: ```suggestion
**1.** Set your output to 1 in the previously mentioned way using `writeDigitalPin(a, 'D13', 1);`. As mentioned earlier, 1 means ON or HIGH and sets the digital pin 13 **D13** to 5V allowing the current flow towards the LED. Check to see if the analog pin 0 **A0** is actually reading an adjustable voltage using `readVoltage(a, 'A0');` in a loop and then use `display(Volts)` to print the value of `Volts` on the "Command Window".
```

```
a = arduino();
writeDigitalPin(a, 'D13', 1)
while(1)
  Volts = readVoltage(a, 'A0');
  display(Volts)
  pause(0.5)
end
```

> The amount assigned to `Volts` should be changing as you turn the potentiometer.

Reviewer 2: ```suggestion
The amount assigned to `Volts` should be changing as you rotate the potentiometer's knob.
```

![Voltage display](assets/Volts.PNG)

> **2.** Set the value of `Volts` as the voltage to be sent through pin 11 using the command: `writePWMVoltage()`. This command is different from `writeDigitalPin()` as its voltage is adjustable between 1 and 5 volts. Whereas the latter is either ON(1), being **5V** or OFF(0), **0V**.

Reviewer 2: ```suggestion
**2.** Set the value of `Volts` as the voltage to be set on digital pin 11 **D11** using the command: `writePWMVoltage()`. This command is adjustable between 0 and 5 volts. It will convert the value to a PWM Signal on the digital pin.
```

> **2.** Set the value of `Volts` as the voltage to be sent through pin 11 using the command: `writePWMVolatge()`. This command is different from `writeDigitalPin()` as it's voltage is adjustable between 1 and 5 volts. Whereas the latter is either ON(1), being **5V** or OFF(0), **0V**.

Reviewer 3: ```suggestion
**2.** Set the value of `Volts` as the voltage to be sent through pin 11 using the command: `writePWMVoltage()`. This command is different from `writeDigitalPin()` as its voltage is adjustable between 1 and 5 volts. Whereas the latter is either ON(1), being **5V** or OFF(0), **0V**.
```

https://www.mathworks.com/help/supportpkg/arduinoio/ref/writepwmvoltage.html

```
a = arduino();
writeDigitalPin(a, 'D13', 1)
while(1)
  Volts = readVoltage(a, 'A0');

>   writePWMVolatge(a, 'D11', Volts)

Reviewer 3: ```suggestion
  writePWMVoltage(a, 'D11', Volts)
```

  display(Volts)
  pause(0.5)
end
```

### Troubleshoot

> If the code is not working, there are some common issues we can troubleshoot:

Reviewer 2: ```suggestion
```

> - Make sure to clear previous Arduino objects linked to the device. You can do this by running the command `clear a` where `a` is the Arduino object.

Reviewer 2: ```suggestion
- If the code get frozen or it seems not working try to clear previous Arduino objects linked to the device. You can do this by running the command `clear <object>`, for example `clear a` will clear the Arduino object.
```

- If you get the message `Unrecognized function or variable 'arduinosetup'` ensure that you have installed the [MATLAB® Support Package for Arduino Hardware](https://www.mathworks.com/matlabcentral/fileexchange/47522-matlab-support-package-for-arduino-hardware).

## Conclusion

> You can now control your output using variable inputs to control your devices and make use of the powerful features for scientific computing and developing engineering applications! Different input modulators similar to the potentiometer processed by the various toolboxes offered my MATLAB can lead to infinite types of ouput beyond that of the brightness of an LED.

Reviewer: ```suggestion
You can now control your output using variable inputs to control your devices and make use of the powerful features for scientific computing and developing engineering applications! Different input modulators similar to the potentiometer processed by the various toolboxes offered my MATLAB can lead to infinite types of output beyond that of the brightness of an LED.
```

## Further Ideas

- You can also use the command line arguments to upload the MATLAB® server to the Arduino® UNO via the `arduino()` command. See more information on using this function in the [MathWorks® documentation](https://www.mathworks.com/help/supportpkg/arduinoio/ref/arduino.html).
- Try writing a MATLAB® code that slowly increases and then decreases the blink speed instead of adjusting the brightness.

> - Use different sensors and parts instead of the potentiometer and LED (maybe a light sensor and a motor?)

Reviewer 2: A motor is not a sensor
```suggestion
- Use different sensors and parts instead of the potentiometer and LED, for example a light sensor.
```

Author: Shall I remove "parts" also?

---

> title: 'Controlling PWM output with a potentiometer on the Arduino® UNO R3 via MATLAB®'

Reviewer 3: ```suggestion
title: 'Controlling PWM Output with a Potentiometer via MATLAB®'
```

Fix content linter failures:

- Make title <=60 characters
  - My thinking is this tutorial is already categorized under the Arduino Uno board section, so that information is available to the reader in other ways
- Use [title case](https://en.wikipedia.org/wiki/Title_case)

difficulty: intermediate
compatible-products: [Arduino Uno Rev 3, Potentiometer]

> description: 'Learn how to dim an LED connceted to the UNO R3 with MATLAB®.'

Reviewer 3: ```suggestion
description: 'Learn how to dim an LED connected to the UNO R3 with MATLAB®.'
```
Fix typo

### Why MATLAB?
MATLAB® is an educational and industrial programming platform used to analyse data, perform simulations and carry out model based designs. Through an interactive communication with an Arduino board, you can expand MATLAB®'s capabilities, while also gaining access to a wide range of math, engineering and plotting functions. Check out the capabilities of MATLAB® over on the [MathWorks website](https://www.mathworks.com/solutions.html#capabilities). In this tutorial, we will establish the connection with MATLAB® via a USB and Wi-Fi connection. The [MATLAB® Support Package for Arduino Hardware](https://www.mathworks.com/matlabcentral/fileexchange/47522-matlab-support-package-for-arduino-hardware) must be installed (which may require a MathWorks® account), in order to facilitate the communication between the MATLAB® software on your computer and your Arduino board. 
 

### LED on the Arduino Uno R3
The internal LED in the UNO R3 is connected to Digital Pin 13. By pulling this pin HIGH, we can turn the LED on and by pulling it LOW we can turn the LED off. But to output a variable voltage we have to use pins such as pin ~11 or ~10.

> ![Pin13](assets/Arduino-Uno-LED.jpg)

Reviewer 3: ```suggestion
![Pin 13](assets/Arduino-Uno-LED.jpg)
```
Fix the "Image doesn't have a (proper) description" content linter error

## Control LED over USB with MATLAB
**1.** Connect your board to the computer via the USB 2.0 Cable Type A/B.

![USB connection to computer](assets/Arduino-Connect.png)

**2.** Open MATLAB and run the command `arduinosetup()` in the Command Window.

![arduino Setup](assets/arduinosetup.png)

**3.** A Graphical User Interface (GUI) appears, that will help you set up the MATLAB®-Arduino connection. To establish the connection over the USB cable, make sure that the USB radio box is selected and then click on Next.

![Selection of Connection Type](assets/connectiontype.png)

**4.** Choose the "UNO" from the dropdown menu, as well as the relevant COM port, and click on the blue `Program` button to upload the Arduino Server to the board. When you have done so, click on next.

> ![Programming-the-arduino](assets/program-Arduino.png)

Reviewer 3: ```suggestion
![Programming the Arduino](assets/program-Arduino.png)
```
Fix the "Image doesn't have a (proper) description" content linter error

**5.** In this step you can review the connection type, Port, board and loaded libraries. You can also click on Test connection to evaluate the Arduino-MATLAB® connection. Next, you should see a Green check-mark signalling the successful connection, as shown in the image below.

> ![Program-successful](assets/programming-done.png)

Reviewer 3: ```suggestion
![Programming successful](assets/programming-done.png)
```
Fix the "Image doesn't have a (proper) description" content linter error

**6.** The UNO R3 is now configured to interact with MATLAB®. Now to create the object in MATLAB®, we run the command `a=arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.

> ![](assets/arduinoproperties.png)

Reviewer 3: ```suggestion
![Arduino properties](assets/arduinoproperties.png)
```
Fix the "Image doesn't have a (proper) description" content linter error

**7.** Enter the command `writeDigitalPin(a,'D13',1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 (D13) is connected to the built-in LED, `a` is the Arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.

***Unlike in the Arduino IDE, here you cannot address the built-in LED using the `LED_BUILTIN` function.***

**8.** Now, you can use this command to continuously blink an LED as part of a `while` loop. Try entering the script below.

```arduino
while (1)
      writeDigitalPin(a, 'D13', 0);
      pause(0.5);
      writeDigitalPin(a, 'D13', 1);
      pause(0.5);
end
```

**9.** See what happens when you change the pause duration. Try replacing the while with an `if` statement to make it blink a certain number of times.

> ## Set up the circuit and code

Reviewer 3: ```suggestion
## Set up the Circuit and Code
```
Use title case
